apps/server/src/tangl/rest/dependencies38.py

```
@@ -27,15 +36,70 @@ def get_service_gateway38() -> ServiceGateway38:
+    api_key: str,
+    *,
+    gateway: ServiceGateway38 | None = None,
+) -> UserAuthInfo:
+    service_gateway = gateway or get_service_gateway38()
+    auth = user_id_by_key(
+        api_key,
+        service_gateway.persistence,
+        reverse_index=_api_key_index38,
+    )
+    if auth is None:
+        raise ValueError("Invalid API key")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)